src/Native/Unix/System.Native/pal_console.c

```diff
@@ -445,7 +445,7 @@ static bool InitializeTerminalCore()
 {
     g_haveInitTermios = tcgetattr(STDIN_FILENO, &g_initTermios) >= 0;
 
-    if (!g_haveInitTermios && errno == ENOTTY)
+    if (!g_haveInitTermios && (errno == ENOTTY || errno == ENODEV))
```

Comment:
@tmds, I'm wondering if we should be even more forgiving here.  What other errors might tcgetattr return? Should we just always set g_noTty if tcgetattr fails?

Reply:
note that I could not reproduce it locally, even if stdin was redirected or executed without tty. 
It was only failing on 10.14 (10.12, 10.13 worked) when executed as service via Launchd/Helix.

Reply:
> What other errors might tcgetattr return?

man pages only list ENOTTY/EBADF.

> Should we just always set g_noTty if tcgetattr fails?

I'm ok with that. I'm also fine with extending this list. No strong preference.

Reply:
I can't see/find the logs. Does the `errno` show up in them clearly? I guess it should, as part of a `Win32Exception`?

Reply:
When it was failing, it did not produce any logs. I had to add custom instrumentation to get:
```
+ /Users/dotnet-bot/dotnetbuild/work/757443a9-b70d-435c-802f-a25debc5f0b9/Payload/dotnet xunit.console.dll System.Net.Ping.Functional.Tests.dll -xml testResults.xml -nocolor -notrait category=nonnetcoreapptests -notrait category=nonosxtests -notrait category=failing -notrait category=OuterLoop -verbose
error: Operation not supported by device
```
I did not see anything propagated back to managed code but the capabilities were limited, perhaps also impacted by debug code. 

I like @stephentoub recommendation and assume if tcgetattr() fails, stdin is not tty.

Reply:
I had expected it to show up from:
https://github.com/dotnet/corefx/blob/355917229c2e5a35b8ea7f14d5d6de2455c5dc1b/src/System.Console/src/System/ConsolePal.Unix.cs#L925-L928

> I like @stephentoub recommendation and assume if tcgetattr() fails, stdin is not tty.

Let's do that.

Reply:
done.